### DISCUSSION TITLE
React 18 support

### DISCUSSION THREAD

**Discussion by akmittal** [twilio-labs/paste#2821] [General] [answered]
For now Paste supports react 16 and 17. Is there plan to support React 18.

**Comment by SiTaggart** [comment]
Hi @akmittal 

Yes, we plan on supporting React 18 soon, we're just working through the details of it right now. There's even an open PR with some of it being worked on right now.

I suspect it'll land in the first part of 2023, maybe Q2, but probably earlier.

There are a lot of dependencies that go into Paste and making sure they all play with React 18 is taking time to make sure we do it right. Ideally, we'd like to keep support for React 16 and 17 after we upgrade, but that might not be possible, so we need to do investigation work first.

Thanks for using Paste! Check back here in couple of months to see how we're progressing.

**Comment by FirstWhack** [comment]
Just chiming in, we need this asap as React 17 does not properly support synthetic events from Shadow Roots. We will be forced to create tech debt and use iframes solely due to Paste not supporting React 18.

**Reply by loreina** [reply]
Hi @FirstWhack, we're currently working on React 18 support and expect support to roll out by the end of Q1 2023. Thanks for your patience!

**Reply by FirstWhack** [reply]
Thank you for the update!

**Comment by nkrantz** [comment, marked as answer]
Hi @akmittal and @FirstWhack - Paste now supports React 18 in [@twilio-paste/core@18.0.0](https://github.com/twilio-labs/paste/releases/tag/%40twilio-paste%2Fcore%4018.0.0)!
